src/utils/datetime.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { getCanonicalLocale } from '@nextcloud/l10n'
+
+const locale = getCanonicalLocale()
+
+const absoluteTimeFormat = new Intl.DateTimeFormat(locale, {
+	year: 'numeric',
+	month: 'long',
+	day: 'numeric',
+	hour: 'numeric',
+	minute: 'numeric',
+})
+// FIXME: Intl.RelativeTimeFormat should use getLanguage(), not getCanonicalLocale()
+const relativeTimeFormat = new Intl.RelativeTimeFormat(locale, {
+	numeric: 'auto',
+})
+
+/**
+ * Format given timestamp in human-readable format (01 January 2025 12:00)
+ * If the unit is not provided, the largest unit is used for rounded duration in milliseconds.
+ * @param {number} timestamp - Timestamp in ms
+ */
+export function formatAbsoluteTime(timestamp) {
```

Comment:
"Absolute time" is a bit different term and rarely used in at least JS ecosystem.
Let's call it `formatTime` or `formatDateTime` to be in line with native function (`Intl.DateTimeFormat`) and popular libraries like dayjs or date-fns